src/common/components/ui/WithTooltip.tsx

```diff
@@ -0,0 +1,10 @@
+import { Tooltip, TooltipProps } from '@patternfly/react-core';
+import React, { ReactElement } from 'react';
+
+export type WithTooltipProps = { tooltipProps?: TooltipProps; children: ReactElement };
+export const WithTooltip: React.FC<WithTooltipProps> = ({ tooltipProps, children }) => {
+  if (tooltipProps === undefined) {
+    return <>{children}</>;
+  }
+  return <Tooltip {...tooltipProps}>{children}</Tooltip>;
```

Comment:
because other components that will use it such as checkboxfield will not need to repeat the logic of 
if (tooltipProps) {} else {}
With your suggestion it doesn't really make a difference, using Tooltip or WithTooltip
We want to support using a field with a tooltip and without